### DISCUSSION TITLE
Looking for favicon example (for ui.run())

### DISCUSSION THREAD

**Discussion by Derf000** [zauberzeug/nicegui#789] [Q&A] [answered]
### Question

Hi
does anyboby could provide me an example of setting custom favicon for ui.run() ?

Using :
- .ico file,
 - emoji
 - svg string (according char_to_data_url(char: str) -> str: in favicon.py)

Thanks ....

**Comment by rodja** [comment, marked as answer]
Emoji is simplest but does not work for every browser:

```py
from nicegui import ui

ui.label("NiceGUI Rocks!")

ui.run(favicon="🚀")
```

If your `.ico` file is located on the filesystem at `static/my_favicon.ico`:

```py
from nicegui import ui

ui.label("NiceGUI Rocks!")

ui.run(favicon="static/my_favicon.ico")
```

SVG is not possible yet. But it's a good idea. We will add that.

**Reply by rodja** [reply]
It is now also available in the online documentation: https://nicegui.io/documentation/run#emoji_favicon

**Comment by Derf000** [comment]
Thank you for the answer, 👍 
My previous attempts did not work (dont know why...) but now it's fine.

It would be great also  if we could pass a Base 64 encoded string to 'favicon' as for ui.image(). [cf. ui.image(f'data:image/png;base64,{base64_string}')]


In order to be able to include the icon directly in code when building with pyinstaller (i know it is possible to --add-data and to deal with sys._MEIPASS to load file but embedbed  image in code could be useful sometimes ...)

**Comment by rodja** [comment]
Was not so difficult to implement. See PR #808. But to make the demos appear on the website is a bit more tricky. Hopefully we can add that.

**Comment by Derf000** [comment]
Thanks for v1.2.10 with favicon "upgrade" works  well.

**Comment by Derf000** [comment]
Hello,
Sorry to bother you again, but since version 1.2.15 the following code no longer works (it was OK until version 1.2.14):

I get an error 36 (filename too long) on a Base64 encoded string as argument for ui.image()

OSError: [Errno 36] File name too long: 'data:image/png;base64,iVBOR ... 5CYII='

Do you have any ideas ?

The code is in the attached zip file.

Thanks in advance

[test_long_b64.zip](https://github.com/zauberzeug/nicegui/files/11733050/test_long_b64.zip)

**Reply by falkoschindler** [reply]
Thanks for spotting this, @gitUser000!
I quickly checked your code and there is, indeed, an issue with base64-encoded images. Because we pass them to `pathlib.Path` to check if a file with that name exists, pathlib complains about the filename being too long. And I guess the specific threshold for the maximum length depends on the OS.
I created PR #1023 to fix this as soon as possible.
